cmd/utils/flags.go

```diff
@@ -2223,7 +2223,8 @@ func MakeGenesis(ctx *cli.Context) *core.Genesis {
 func MakeChain(ctx *cli.Context, stack *node.Node) (chain *core.BlockChain, chainDb ethdb.Database) {
 	var err error
 	chainDb = MakeChainDatabase(ctx, stack, false) // TODO(rjl493456442) support read-only database
-	config, _, err := core.SetupGenesisBlock(chainDb, MakeGenesis(ctx), false)
+	gpec := MakeGenesis(ctx)
+	config, err := core.LoadChainConfig(chainDb, gpec)
```

Comment:
hmm why we need the change here ?

Reply:
Same logic in `backend`, there is a line below that calls `core.NewBlockChain`:
```
chain, err = core.NewBlockChain(chainDb, cache, gpec, nil, engine, vmcfg, nil, nil)
```